core/hash_map.h

```diff
@@ -230,7 +230,7 @@ class HashMap {
 		if (!p_t.hash_table || p_t.hash_table_power == 0)
 			return; /* not copying from empty table */
 
-		hash_table = memnew_arr(Element *, 1 << p_t.hash_table_power);
+		hash_table = memnew_arr(Element *, 1ULL << p_t.hash_table_power);
```

Comment:
If we're sticking with the type casting, this should probably be `(uint64_t)1` as well.